pkg/bearertoken/grpc.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package bearertoken
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const Key = "bearer.token"
+
+type tokenatedServerStream struct {
+	grpc.ServerStream
+	context context.Context
+}
+
+// getValidBearerToken attempts to retrieve the bearer token from the context.
+// It does not return an error if the token is missing.
+func getValidBearerToken(ctx context.Context, bearerHeader string) (string, error) {
+	bearerToken, ok := GetBearerToken(ctx)
+	if ok && bearerToken != "" {
+		return bearerToken, nil
+	}
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return "", nil
+	}
+
+	return tokenFromMetadata(md, bearerHeader)
+}
+
+// tokenFromMetadata extracts the bearer token from the metadata.
+func tokenFromMetadata(md metadata.MD, bearerHeader string) (string, error) {
```

Comment:
unnecessary function
```suggestion
```

Reply:
its only called once , so i should refactor this